application/src/main/resources/org/opentripplanner/apis/gtfs/schema.graphqls

```diff
@@ -4603,6 +4603,42 @@ input TransferPreferencesInput {
   slack: Duration
 }
 
+"A collection of selectors for what routes/agencies should be included in or excluded from the search."
+input TransitFilterInput {
+  """
+  A list of selectors for what routes/agencies should be allowed during search.
+    
+  In order to be excluded a route/agency has to match with at least one selector.
+    
+  An empty list or a list containing `null` is forbidden.
+  """
+  exclude: [TransitFilterSelectInput!]
+  """
+  A list of selectors for what routes/agencies should be excluded during the search.
+    
+  If route/agency matches with at least one selector it will be included.
+    
+  An empty list or a list containing `null` is forbidden.
+  """
+  include: [TransitFilterSelectInput!]
+}
+
+"""
+A list of selectors for filter including or excluding entities from the routing results. Null
+means everything is allowed to be returned and empty lists a not allowed.
+
+An entity will match with selectors if it matches with all non-empty lists.
```

Comment:
When we just have agencies and routes to select, this model is a bit clumsy. I think we could use a `@oneOf` here until we include something where combining multiple fields makes sense. I don't think removing the `@oneOf` restriction is a breaking change (but I'm not 100% sure).

Reply:
I think it helps people to make a logic mistake so I've added it.

I, too, believe it would not be a breaking change to remove it, so lets add it.